app/Services/MenuService/SidebarMenuService.php

```diff
@@ -0,0 +1,275 @@
+                    'permission' => 'pulse.view'
+                ]
+            ],
+            'priority' => 40,
+            'permission' => ['pulse.view', 'actionlog.view']
+        ]);
+
+
+        $this->addMenuItem([
+            'label' => __('Settings'),
+            'icon' => 'settings.svg',
+            'id' => 'settings-submenu',
+            'active' => \Route::is('admin.settings.*') || \Route::is('admin.translations.*'),
+            'children' => [
+                [
+                    'label' => __('General Settings'),
+                    'route' => route('admin.settings.index'),
+                    'active' => \Route::is('admin.settings.index'),
+                    'priority' => 20,
+                    'permission' => 'settings.edit'
+                ],
+                [
+                    'label' => __('Translations'),
+                    'route' => route('admin.translations.index'),
+                    'active' => \Route::is('admin.translations.*'),
+                    'priority' => 10,
+                    'permission' => ['translations.view', 'translations.edit']
+                ]
+            ],
+            'priority' => 1,
+            'permission' => ['settings.edit', 'translations.view']
+        ], 'Settings');
+
+
+        $this->addMenuItem([
+            'label' => __('Logout'),
+            'icon' => 'logout.svg',
+            'route' => route('logout'),
+            'active' => false,
+            'id' => 'logout',
+            'priority' => 1,
+            'html' => '
+                <li class="hover:menu-item-active">
+                    <form method="POST" action="' . route('logout') . '">
+                        ' . csrf_field() . '
+                        <button type="submit" class="menu-item group w-full text-left menu-item-inactive text-black dark:text-white hover:text-black">
+                            <img src="' . asset('images/icons/logout.svg') . '" alt="Logout" class="menu-item-icon dark:invert">
+                            <span class="menu-item-text">' . __('Logout') . '</span>
+                        </button>
+                    </form>
+                </li>
+            '
+        ], 'Settings');
+
+        // Sort each group by priority (lower first)
+        foreach ($this->groups as &$groupItems) {
+            usort($groupItems, function ($a, $b) {
+                return $a->toArray()['priority'] <=> $b->toArray()['priority'];
+            });
+        }
+
+        // Add filters so that developers can modify the menu
+        $result = [];
+        foreach ($this->groups as $group => $items) {
+            $menuArr = array_map(function ($item) {
+                return $item->toArray();
+            }, $items);
+            $result[$group] = ld_apply_filters('sidebar_menu_' . strtolower($group), $menuArr);
+        }
+
+        return $result;
+    }
+
+    public function render($menus, $textColorVar = 'textColor', $submenusVar = 'submenus')
```

Comment:
use proper variable name comment so typehinting works how I showed in the video.